sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/OkHttpClientProvider.java

```diff
@@ -13,4 +14,18 @@
     public HttpClient createInstance() {
         return new OkHttpAsyncHttpClientBuilder().build();
     }
+
+    @Override
+    public HttpClient createInstance(HttpClientOptions clientOptions) {
+        OkHttpAsyncHttpClientBuilder builder = new OkHttpAsyncHttpClientBuilder();
+
+        if (clientOptions != null) {
+            builder = builder.proxy(clientOptions.getProxyOptions())
+                .configuration(clientOptions.getConfiguration())
+                .writeTimeout(clientOptions.getWriteTimeout())
+                .readTimeout(clientOptions.getReadTimeout());
```

Comment:
If I understand correctly `ClientOptions::responseTimeout` is not applicable for OkHttp right?

Reply:
Correct, OkHttp doesn't have an explicit API to set the timeout period for receiving a response for a request.